code/modules/atmospherics/machinery/components/binary_devices/temperature_pump.dm

```diff
@@ -0,0 +1,80 @@
+/obj/machinery/atmospherics/components/binary/temperature_pump
+	icon_state = "tpump_map-3"
+	name = "temperature pump"
+	desc = "A pump that moves heat only."
+
+	can_unwrench = TRUE
+	shift_underlay_only = FALSE
+
+	var/heat_transfer_rate = 0
+	var/max_heat_transfer_rate = 4500
+
+	construction_type = /obj/item/pipe/directional
+	pipe_state = "tpump"
+
+/obj/machinery/atmospherics/components/binary/temperature_pump/CtrlClick(mob/user)
+	if(can_interact(user))
+		on = !on
+		investigate_log("was turned [on ? "on" : "off"] by [key_name(user)]", INVESTIGATE_ATMOS)
+		update_icon()
+	return ..()
+
+/obj/machinery/atmospherics/components/binary/temperature_pump/AltClick(mob/user)
+	if(can_interact(user))
```

Comment:
You should check if the rate is already maxed to prevent spamming the investigation log.